packages/editor/src/components/provider/disable-non-page-content-blocks.js

```diff
@@ -2,54 +2,52 @@
  * WordPress dependencies
  */
 import { useSelect, useDispatch } from '@wordpress/data';
-import {
-	useBlockEditingMode,
-	store as blockEditorStore,
-} from '@wordpress/block-editor';
+import { store as blockEditorStore } from '@wordpress/block-editor';
 import { useEffect } from '@wordpress/element';
 
-/**
- * Internal dependencies
- */
-import { PAGE_CONTENT_BLOCK_TYPES } from './constants';
+const PAGE_CONTENT_BLOCKS = [
+	'core/post-title',
+	'core/post-featured-image',
+	'core/post-content',
+];
+
+function useDisableNonPageContentBlocks() {
+	const contentIds = useSelect( ( select ) => {
```

Comment:
Just a question for my own understanding: it looks like this is safely not creating a new reference too frequently (i.e. it's used in the `useEffect`'s dependencies and doesn't cause any issues). I'm used to the idea of trying to avoid `.filter()` calls within a `useSelect` because we then wind up with a new reference. In this case, is the reason that it's safe that `contentIds` is the value returned by the callback here, rather than destructuring a value within an object?

Reply:
Correct, this is a "hack"; `mapSelect` returns an array of `clientId` strings, which can correctly pass shallow equality check.

> I'm used to the idea of trying to avoid .filter() calls within a useSelect because we then wind up with a new reference.

Reply:
Ah, gotcha. Thanks for confirming!

Reply:
Huh, I didn't know it was a "hack" 😀 I just prefer not returning multiple things from `useSelect` where possible as I think it's easier to read.

There's a warning if you do it wrong so I just let the computer do the thinking about this stuff.

https://github.com/WordPress/gutenberg/blob/8f61b9ba5e1e898626a6b94f52411222e3c2f463/packages/data/src/components/use-select/index.js#L163-L165

Reply:
I mean “hack” because it’s not obvious why it’s okay to use `filter` in current situation if you’re not familiar with `useSelect` internals.